### PR TITLE
[SGTM multi-org support] modify github auth helpers to take in github_org_name from payload and save it to memory

### PR DESCRIPTION
**Pull request by suzyng83209**
### Summary

- Modifies the sgtm_github_auth variable to a getter since we need org_name which is only receivable via webhook
- adds a MutableMapping[str, SGTMGithubAuth] to save and retrieve auth classes depending on org name
- updates all of the clients to use the new auth

Asana tasks:
https://app.asana.com/0/1149418478823393/1205088646274292/f

Relevant deployment: 

CC: @harshita-gupta @prebeta @suzyng83209 @vn6 @michael-huang87

### Test Plan

- [x] staging deploy works with Asana repo: [TestingSGTM task was re-opened](https://app.asana.com/0/1207118214445707/1207984117265636) after its PR was re-opened https://github.com/Asana/testing-sgtm/pull/16 and assignee removed 
- [x] staging deploy works with Asana-ITSE repo

### Risks

Pull Request synchronized with [Asana task](https://app.asana.com/0/0/1208341767512029)
Pull Request: https://github.com/Asana/SGTM/pull/201

### CODE REVIEW

**Comment by suzyng83209** [comment]
* **#201** <a href="https://app.graphite.dev/github/pr/Asana/SGTM/201?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @suzyng83209 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by harshita-gupta — APPROVED** [review]
Auth bits look ok, someone on CI should probably approve SGTM logic changes!

File: src/github/client.py [hunk 1]
```diff
@@ -5,11 +5,9 @@
 from src.github.get_app_token import sgtm_github_auth
 from src.logger import logger
 
-gh_client = sgtm_github_auth.get_rest_client()
-
 
 def _get_pull_request(owner: str, repository: str, number: int) -> PullRequest:  # type: ignore
-    repo = gh_client.get_repo(f"{owner}/{repository}")
+    repo = sgtm_github_auth(owner).get_rest_client().get_repo(f"{owner}/{repository}")
```
Comment (vn6) [inline src/github/client.py:10]: oh hmm does this mean we're creating a new client for every request now

**Review by vn6 — APPROVED** [review]
lgtm

File: src/github/client.py (hunk 1 above)
Comment (suzyng83209) [reply src/github/client.py:10]: no the `sgtm_github_auth` method accesses a map of `github_org_name -> github_auth_client` to create and/or retrieve an existing client to use.

Comment (suzyng83209) [reply src/github/client.py:10]: oh actually, we are creating a new auth for every test run so I'll make this more robust for local auth

**Pull request merged**